### ISSUE TITLE
Match pattern with `@` accepted although to few arguments mentioned

### ISSUE THREAD

**Issue by fakusb** [coq/coq#13534]
#### Description of the problem
Is it intended that trailing implicit arguments can be ommited in  patterns when using `@`?
I would expect the following to fail at both `Check`s:

```` coq
Inductive Test := test : forall {a b c d e f : bool}, Test.
About test.
Axiom t : Test.

Check match t with | @test => _ end.
Check match t with | @test _ => _ end.
````

(This reminds me of https://github.com/mattam82/Coq-Equations/issues/77 and the confusion I had there, and might mean that Equations also allows strange things, which might sometimes lead to the same confusion for new users I had there. But I did not test if this transfers to Equations).

(I found this while constructing the example for #13533 )

#### Coq Version

master from a few days ago ( 0501761 )

**Comment by Alizter** [comment]
cc @herbelin

**Comment by SkySkimmer** [comment]
Looks fixed on current master

**Comment by herbelin** [comment]
Apparently, it is #12100 (in v8.14) which fixed it. There was no changelog because I did not realize this was this bug report.

**Comment by mrhaandi** [comment]
Closed by #12100.
